### PR TITLE
Use JavaBeans naming standard to guess fields name

### PR DESCRIPTION
**Pull request by pcmiel**

### CODE REVIEW

File: src/main/java/com/futureprocessing/documentjuggler/commons/FieldNameExtractor.java [hunk 1]
```diff
@@ -10,6 +12,24 @@
...
         DbField field = from(method).read(DbField.class);
-        return field.value();
+        if (field != null) {
+            return field.value();
+        }
+
+        List<String> toRemove = Arrays.asList("get", "set", "with");
+
+        String methodName = method.getName();
+        for (String remove : toRemove) {
```
Comment (szymonsobocik) [inline src/main/java/com/futureprocessing/documentjuggler/commons/FieldNameExtractor.java]: Suggestion: change `remove` to `prefix`

File: src/test/java/com/futureprocessing/documentjuggler/fieldNames/FieldNameTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+        Movie withTitle(String title);
+
+        String title();
+
+        String getLongNameWithUppercases();
+    }
+
+    @Test
+    public void shouldCreateDbFieldNameFromGetterName() throws NoSuchMethodException {
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/fieldNames/FieldNameTest.java]: `shouldGet` instead `shouldCreate`

File: src/test/java/com/futureprocessing/documentjuggler/fieldNames/FieldNameTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        //given
+        Method method = Movie.class.getMethod("getTitle");
+
+        //when
+
+        //then
+        assertThat(FieldNameExtractor.getFieldName(method)).isEqualTo("title");
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/fieldNames/FieldNameTest.java]: `FieldNameExtractor.getFieldName(method)`  should be in `//when` section

File: src/test/java/com/futureprocessing/documentjuggler/fieldNames/FieldNameTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        Method method = Movie.class.getMethod("getLongNameWithUppercases");
+
+        //when
+
+        //then
+        assertThat(FieldNameExtractor.getFieldName(method)).isEqualTo("longNameWithUppercases");
+    }
+
+    @Test
+    public void shouldntCreateDbFieldNameFromMethodName() throws NoSuchMethodException {
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/fieldNames/FieldNameTest.java]: Please write explicit `Not`  ie.: `shouldNot....`  so it'll be more visible.

File: src/test/java/com/futureprocessing/documentjuggler/fieldNames/ValidateFiledsTest.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import com.futureprocessing.documentjuggler.commons.Validator;
+import com.futureprocessing.documentjuggler.exception.validation.UnknownFieldException;
+import org.junit.Test;
+
+import java.lang.reflect.Method;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.fail;
+
+public class ValidateFiledsTest {
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/fieldNames/ValidateFiledsTest.java]: typo

File: src/test/java/com/futureprocessing/documentjuggler/fieldNames/ValidateFiledsTest.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Test
+    public void shouldBeValidInGetterSetterWithIfDbFieldAnnotationNotExist() throws NoSuchMethodException {
+        //given
+        Method method = Movie.class.getMethod("getTitle");
+
+        //when
+        //then
+        Validator.validateField(method);
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/fieldNames/ValidateFiledsTest.java]: I think this should be in `//when` section.

Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/fieldNames/ValidateFiledsTest.java]: And maybe it'll be worth to wrap this into try/catch block, and explicitly ensure that no exception was thrown.

File: src/test/java/com/futureprocessing/documentjuggler/fieldNames/ValidateFiledsTest.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        //when
+        try {
+            Validator.validateField(method);
+        } catch (UnknownFieldException ex) {
+            // then
+            assertThat(ex.getMethod()).isEqualTo(Movie.class.getMethod("title"));
+            return;
+        }
+
+        fail();
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/fieldNames/ValidateFiledsTest.java]: Use `JugglerAssertions.failExpectedException()`.

File: src/test/java/com/futureprocessing/documentjuggler/fieldNames/FieldNameTest.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+import com.futureprocessing.documentjuggler.annotation.CollectionName;
+import com.futureprocessing.documentjuggler.commons.FieldNameExtractor;
+import org.junit.Test;
+
+import java.lang.reflect.Method;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FieldNameTest {
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/fieldNames/FieldNameTest.java]: Should be `FieldNameExtractorTest`

File: src/test/java/com/futureprocessing/documentjuggler/fieldNames/ValidateFiledsTest.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Test
+    public void shouldThrowUnknownFieldExceptionIfOneOfMethodsIsNotAnnotatedWithDbField() throws NoSuchMethodException {
+        //given
+        Method method = Movie.class.getMethod("title");
+
+        //when
+        try {
+            Validator.validateField(method);
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/fieldNames/ValidateFiledsTest.java]: Discussed with Bartek.
We should remove `Validator.validateField()` method completely, and move this validation to `FieldNameExtractor.getFieldName()`.

**Comment by exys666** [comment]
+1

**Comment by szymonsobocik** [comment]
@pcmiel Please merge this pull request, and fix tests: https://github.com/FutureProcessing/document-juggler/pull/68

**Pull request merged**